### ISSUE TITLE
Cambiar criterio de operador activo

### ISSUE THREAD

**Issue by GioSunrider** [asrob-uc3m/operadores#58]
Debido a que @MrTommyGuns esta cerca de expirar su "licencia" y la mia hace mucho que expiró, ademas que ambos somos los actuales responsables del grupo y de las impresoras, propongo un cambio en el sistema de los criterios.
El cambio seria que a los responsables vigentes del grupo Printer y de las impresoras, no se les caduque al año, ya que somos los que posiblemente mejor conocemos las impresoras y el estado en el que se encuentran. Ademas seguimos realizando formaciones durante alguna de las reuniones a los integrantes de los grupos de la asocion.

Para realizar el cambio creo que solo habria que añadir una linea como la que tiene asrobuc3m en [operadores.py](https://github.com/asrob-uc3m/operadores/blob/master/operadores.py) vease el ejemplo.

`   operator=dict()
    operator['raw_formed_date'] = "2011-06-03T09:00:00Z"
    operator['formed_in_issue_url'] = "http://www.reprap.org/wiki/Clone_Wars:_historia/es"
    operator['username'] = "asrobuc3m"
    operator['n_of_operators_formed'] = 0
    operator['authorized'] = 1 #Always
    operator['msg'] = "User with login asrobuc3m is always authorized"
    unsorted_operators[operator['username']] = operator`

**Comment by jgvictores** [comment]
Casi un duplicado de otra issue donde la propuesta es de un sistema de 4 años: https://github.com/asrob-uc3m/operadores/issues/49

Técnicamente no está recopilado en [estatutos](https://github.com/asrob-uc3m/estatutos) ([ahora mismo](https://github.com/asrob-uc3m/estatutos/tree/b681f4e7ce8bdcfee5069526f4d8fa4927efcd45)), pero como afecta muy horizontal, quizás merezca la pena una votación con diferentes alternativas.

PD1: Personalmente lo de `los responsables vigentes del grupo Printer y de las impresoras, no se les caduque al año, ya que somos los que posiblemente mejor conocemos las impresoras` me parece razonable.

PD2: @GioSunrider Edité tu descripción de issue para que fuese un bloque de código Python con syntax highlighting (<https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks#syntax-highlighting>).

**Comment by GioSunrider** [comment]
Cierto, sabia que esta cuestion no era la primera vez que pasaba. Dejo eta como continuacion de la otra ya que esta creo que tiene mas informacion y es mas propuesta.

**Comment by GioSunrider** [comment]
Sin problema por el cambio, de hecho gracias era la intencion que tenia pero no sabia como hacerlo.
Y continuamos en esta ya que tiene mas calidad de propuesta de cambio.

**Comment by David-Estevez** [comment]
Voy a repostear aquí mi comentario en la otra issue, porque creo que sigue siendo válido:

> No veo el problema de que esa gente necesite validación. Validación no significa tener que volver a hacer la formación, sino que necesitan que alguien que esté en activo dé fé de que esa formación sigue siendo válida y los conocimientos que tienen esos operaciones siguen vigentes. Sólo en caso de que los conocimientos no sirvan habría que volver a formarles.
> 
> El único problema que veo es que se os pase a todos el pedir la validación y no haya ningún operador validado para validar al resto... :joy:

Solventar el último punto es tan sencillo como ponerse un recordatorio en algún lado para que no se te pase la fecha.

**Comment by GioSunrider** [comment]
Y sigo pensando que esto: 
> El único problema que veo es que se os pase a todos el pedir la validación y no haya ningún operador validado para validar al resto... 😂
Es el mayor de los problemas, porque ahora lo tenemos fresco pero en un futuro puede que no haya nadie, de ahi mi propuesta de que ambos responsables sean fijos para siempre como asrobuc3m y variales en el tiempo de tal manera que asi siempre haya alguien real y no figurativo.

**Comment by David-Estevez** [comment]
Si no hay nadie, tampoco están los responsables, así que lo que dices no me tiene mucho sentido...
Lo de "fijos para siempre y variables en el tiempo" también me ha roto un poco las neuronas...

¿Tan complicado es poner un recordatorio para renovarlo, igual que renovamos la matrícula de la uni cada año o el DNI cuando está a punto de caducarse?

**Comment by MrTommyGuns** [comment]
Si realmente es solo una especie de revisión de que tienes las habilidades/conceptos sin necesidad de hacer nada en concreto, las personas que trabajan en el grupo de Printer One  (en el que todos o casi todos somos operadores) realmente quedaría renovado "automáticamente" ya que por lo que entiendo, hay otros operadores a tu al rededor y además se están imprimiendo, ajustando y arreglando piezas continuamente...
Otro tema sería para las personas que lo hacen para imprimir por su cuenta o trabajar desde otro grupo de trabajo en el que no hay nadie "autorizado" (aka otro operador) que pueda dar fe de sus conocimientos.

**Comment by MrTommyGuns** [comment]
Por otro lado, una pregunta por ignorancia total del asunto.
No podría programarse un correo electrónico al año de la formación que llegara de forma automática al correo del operador al cual se le ha caducado? O algo del estilo?
En plan "tu licencia de operador de impresoras 3D en asrob ha caducado, por favor si sigue interesado en usar nuestras impresoras póngase en contacto con nosotros para actualizarlo".

**Comment by JorFru** [comment]
> El cambio seria que a los responsables vigentes del grupo Printer y de las impresoras, no se les caduque al año,
esto me parece una chapuza.

Mi idea es que la formación está vigente un año desde la última vez que se hizo uso de las impresoras, no desde que se recibió la formación.

**Comment by jgvictores** [comment]
IMHO, lo ideal desde el punto de vista teórico: Establecer una especie de política de versiones (desde el punto de vista de operador de impresora). Una formación te habilitaría para `1.x`. Esto va de la mano de mejoras incementales de las impresoras `1.0`, `1.1`, ...`1.27`... que sube de versión cuando se rompe la compatibilidad hacia atrás desde el punto de vista de operario y se empieza con `2.0` (y cada operario se debe formar en `2.x`).

Entiendo que es muy difícil desde el punto de vista de implementación en la práctica, y que habría que tener todo siempre muy bien documentado, por lo que dejo simplemente por si surgen nuevas ideas.

**Comment by David-Estevez** [comment]
> Mi idea es que la formación está vigente un año desde la última vez que se hizo uso de las impresoras, no desde que se recibió la formación.

La formación está vigente un año desde la última formación. Al llegar al año, simplemente se revisa:

1. Que la persona sigue interesada en ser operador, y por tanto hace un esfuerzo activo por renovar su estatus de operador.
2. Que las impresoras no han cambiado lo suficiente en ese tiempo como para que la formación de esa persona haya quedado obsoleta, en cuyo caso con un "curso de reciclaje" con los cambios sería suficiente.

Lo ideal sería algo como lo que dice Juan, pero estoy de acuerdo con él en que implementar eso en la vida real es un pollo de dimensiones bíblicas.
